Yank/restraints.py

```diff
@@ -766,89 +1168,367 @@ class Boresch(OrientationDependentRestraint):
-                    reject = True
+        Parameters
+        ----------
+        thermodynamic_state : openmmtools.states.ThermodynamicState
+            The thermodynamic state.
 
-            r0 = self._parameters['r_aA0']
-            K = self._parameters['K_r']
-            sigma = unit.sqrt(NSIGMA * kT / (K/2.))
-            if (r0 < sigma):
-                logger.debug('Reject because r_aA0 is too close to 0 for standard state correction to be accurate.')
-                reject = True
+        Returns
+        -------
+        DeltaG : float
+           Computed standard-state correction in dimensionless units (kT).
+
+        """
+        # Check if all parameters are defined.
+        self._check_parameters_defined()
+
+        # Shortcuts variables.
+        pi = np.pi
+        kT = thermodynamic_state.kT
+        p = self  # For the parameters.
+
+        # Eq 32 of Ref [1]. Multiply by unit.radian**5 to remove the
+        # expected unit value radians is a soft unit in this case, it
+        # cancels in the math, but not in the equations above.
+        DeltaG = -np.log(
+            (8. * pi ** 2 * V0) / (p.r_aA0 ** 2 * unit.sin(p.theta_A0) * unit.sin(p.theta_B0)) *
+            unit.sqrt(p.K_r * p.K_thetaA * p.K_thetaB * p.K_phiA * p.K_phiB * p.K_phiC) / (2 * pi * kT) ** 3 *
+            unit.radian**5
+        )
+
+        return DeltaG
 
-            attempt += 1
+    def _get_standard_state_correction_numerical(self, thermodynamic_state):
+        """Return the standard state correction using the numerical method.
 
-    def _is_collinear(self, positions, atoms, threshold=0.9):
-        """Report whether any sequential vectors in a sequence of atoms are collinear to within a given dot product threshold.
+        Uses numerical integral to the partition function contriubtions for
+        r and theta, analytical for phi
 
         Parameters
         ----------
-        positions : simtk.unit.Quantity of natoms x 3 with units compatible with nanometers
-            Reference positions to use for imposing restraints
-        atoms : array-like, dtype:int, length:natoms
-            A seris of indices of atoms
+        thermodynamic_state : openmmtools.states.ThermodynamicState
+            The thermodynamic state.
+
+        Returns
+        -------
+        DeltaG : float
+           Computed standard-state correction in dimensionless units (kT).
+
+        """
+        # Check if all parameters are defined.
+        self._check_parameters_defined()
+
+        def strip(passed_unit):
+            """Cast the passed_unit into md unit system for integrand lambda functions"""
+            return passed_unit.value_in_unit_system(unit.md_unit_system)
+
+        # Shortcuts variables.
+        pi = np.pi
+        kT = thermodynamic_state.kT
+        p = self  # For the parameters.
+
+        # Radial
+        sigma = 1 / unit.sqrt(p.K_r / kT)
```

Comment:
Good idea! I checked on the interpreter and `unit.sqrt` raises a `ValueError("math domain error")` when called with a negative argument, so I'm tempted to leave the handling of eventual errors to it. The error trace is pretty clear and this way I won't have to add a bunch of `assert` statements.